docker-startup.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+
+# This script is used to start the import of kosmtik containers for the Docker development environment.
+# You can read details about that in SETUP.md
+
+# Testing if database is ready
+i=1
+MAXCOUNT=60
+echo "Waiting for PostgreSQL to be running"
+while [ $i -le $MAXCOUNT ]
+do
+  pg_isready -q && echo "PostgreSQL running" && break
+  sleep 2
+  i=$((i+1))
+done
+test $i -gt $MAXCOUNT && echo "Timeout while waiting for PostgreSQL to be running"
+
+case "$1" in
+import)
+  # Creating default database
+  psql -c "SELECT 1 FROM pg_database WHERE datname = 'gis';" | grep -q 1 || createdb gis && \
+  psql -d gis -c 'CREATE EXTENSION IF NOT EXISTS postgis;' && \
+  psql -d gis -c 'CREATE EXTENSION IF NOT EXISTS hstore;'
+
+  # Creating default import settings file editable by user and passing values for osm2pgsql
+  if [ ! -e ".env" ]; then
+    cat > .env <<EOF
+# Environment settings for importing to a Docker container database
+PG_WORK_MEM=${PG_WORK_MEM:-16MB}
+PG_MAINTENANCE_WORK_MEM=${PG_MAINTENANCE_WORK_MEM:-256MB}
+OSM2PGSQL_CACHE=${OSM2PGSQL_CACHE:-512}
+OSM2PGSQL_NUMPROC=${OSM2PGSQL_NUMPROC:-1}
+OSM2PGSQL_DATAFILE=${OSM2PGSQL_DATAFILE:-data.osm.pbf}
+EXTERNAL_DATA_SCRIPT_FLAGS=${EXTERNAL_DATA_SCRIPT_FLAGS:-}
```

Comment:
This will set the environment variables to fixed values, if not overridden at the time of running this script. Did you intend to do the evaluation and default parameter usage when the env script is actually executed? In this case you need to escape the ```$``` here.

Is ```${something:-value}``` in POSIX so it will work with any /bin/sh?

Reply:
Hmm I copied this from the same script in openstreetmap-carto, https://github.com/gravitystorm/openstreetmap-carto/blob/master/scripts/docker-startup.sh#L27-L35. 

So this will only set the initial values to the defaults, and then create the file if it does not exist. 

Then, the values will be used during the import, like documented here: https://github.com/gravitystorm/openstreetmap-carto/blob/62e8d547a39f8b40e774e98d295b4f9758bc41be/DOCKER.md?plain=1#L59-L75

The environment variables set in runtime will always override the environment variables from `.env`.

Reply:
> Is ${something:-value} in POSIX so it will work with any /bin/sh?

Yes, I found https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02 which documents `${something:-value}` for using a default value